src/Pim/Bundle/CatalogBundle/Doctrine/Common/Saver/ProductSaver.php

```diff
@@ -115,4 +112,50 @@ protected function validateProduct($product)
             );
         }
     }
+
+    /**
+     * Calculates current product completenesses.
+     *
+     * The current completenesses collection is first cleared, then newly calculated ones are set to the product.
+     *
+     * @param ProductInterface $product
+     */
+    protected function calculateProductCompletenesses(ProductInterface $product)
+    {
+        $completenessesCollection = $product->getCompletenesses();
+
+        if (!$completenessesCollection->isEmpty()) {
+            $this->dropCompletenesses($completenessesCollection);
+        }
+
+        $newCompletenesses = $this->completenessCalculator->calculate($product);
+
+        foreach ($newCompletenesses as $completenessesByChannel) {
```

Comment:
Because you don't have necessarily the same values in all the completenesses. For instance: non scopable/non localizable attributes have corresponding values in all completenesses, but a scopable/localizable only in one.